File: Java/No2AddTwoNumbers.java
class No2AddTwoNumbers {
  class ListNode {
      int val;
      ListNode next;
      ListNode(int x) { val = x; }
  }
  public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
      ListNode dummy = new ListNode(-1);
      ListNode p = dummy;
      int flag = 0;
      while(l1!=null || l2!=null || flag != 0 ) {
        int val1 = l1 == null ? 0 : l1.val;
        int val2 = l2 == null ? 0 : l2.val;
        int temp = val1 + val2 + flag;
        flag = temp / 10;
        p.next = new ListNode(temp % 10);
        p = p.next;
        l1 = l1 == null ? null : l1.next;
        l2 = l2 == null ? null : l2.next;
      }
      return dummy.next;
  }
}

File: Kotlin/No2AddTwoNumbers.kt
class No2AddTwoNumbers {
    class ListNode(var `val`: Int) {
        var next: ListNode? = null
    }
    fun addTwoNumbers(l1: ListNode?, l2: ListNode?): ListNode? {
        val dummy = ListNode(-1)
        var p = dummy
        var flag = 0
        var p1 = l1
        var p2 = l2
        while (p1 != null || p2 != null ||flag != 0) {
            val val1 = p1?.`val` ?: 0
            val val2 = p2?.`val` ?: 0
            val temp = val1 + val2 + flag
            flag = temp / 10
            p.next = ListNode(temp % 10)
            p = p.next
            p1 = p1?.next
            p2 = p2?.next
        }
        return dummy.next
    }
}